test/com/google/enterprise/adaptor/database/TupleReaderTest.java

```diff
@@ -129,34 +129,48 @@ public void testVarchar() throws Exception {
   }
 
   @Test
-  public void testVarchar_xml() throws Exception {
-    final String template = ""
+  public void testInteger_null() throws Exception {
+    executeUpdate("create table data(colname integer)");
+    executeUpdate("insert into data(colname) values(null)");
+    final String golden = ""
         + "<database>"
         + "<table>"
         + "<table_rec>"
-        + "<COLNAME SQLType=\"VARCHAR\">%s</COLNAME>"
+        + "<COLNAME SQLType=\"INTEGER\" ISNULL=\"true\"/>"
         + "</table_rec>"
         + "</table>"
         + "</database>";
-    String xml = String.format(template, "onevalue");
-    executeUpdate("create table data(colname varchar)");
-    executeUpdate("insert into data(colname) values('" + xml + "')");
     ResultSet rs = executeQueryAndNext("select * from data");
     String result = generateXml(rs);
-    assertEquals(
-        String.format(template, xml.replace("<", "&lt;").replace(">", "&gt;")),
-        result);
+    assertEquals(golden, result);
   }
 
   @Test
-  public void testVarchar_null() throws Exception {
-    executeUpdate("create table data(colname varchar)");
+  public void testBoolean() throws Exception {
```

Comment:
Below is right, but I was of a different mind when I didn't change the prod code. I had suggested putting these first in prod, but the canonical order has them in a second group (really third, after integers and reals). I'll change prod.